src/daft-recordbatch/src/ops/window.rs

```diff
@@ -127,19 +130,102 @@ impl RecordBatch {
             }
         };
 
-        self.window_agg_rows_between(
-            agg_expr,
-            &name,
-            dtype,
-            start_boundary,
-            end_boundary,
-            min_periods,
-            total_rows,
-        )
+        match agg_expr {
+            AggExpr::Sum(_)
+            | AggExpr::Count(..)
+            | AggExpr::Mean(_)
+            | AggExpr::Min(_)
+            | AggExpr::Max(_)
+            | AggExpr::CountDistinct(_) => self.window_agg_rows_optimized(
```

Comment:
I notice you have this match but also one in `create_window_agg_state`. Could we consolidate them?